arch/risc-v/src/fe310/fe310_irq_dispatch.c

```diff
@@ -52,10 +52,10 @@ volatile uint32_t * g_current_regs;
  * fe310_dispatch_irq
  ****************************************************************************/
 
-void *fe310_dispatch_irq(uint32_t vector, uint32_t *regs)
+void *fe310_dispatch_irq(uint32_t vector, uintptr_t *regs)
 {
   uint32_t  irq = (vector >> 27) | (vector & 0xf);
-  uint32_t *mepc = regs;
+  uint32_t *mepc = (uint32_t *)regs;
```

Comment:
Maybe can switch to `uintptr_t` here at well? Or it is not possible?

Reply:
The same for `vector` and `irq`

Reply:
Yes, all of these can be replaced by uintptr_t

Reply:
```suggestion
  uintptr_t *mepc = regs;
```